### ISSUE TITLE
Redundant nztm coordinate transformation functions

### ISSUE THREAD

**Issue by lispandfound** [ucgmsim/qcore#306]
Once #303 is merged, there will be two ways to convert coordinates, `wgs_nztm2000x` and `coordinates.wgs_depth_to_nztm`. This old function is used in two places:

1. https://github.com/ucgmsim/mapping/blob/deda5ebb649316ef91cca53b83a5909914017953/mapbox/vs30/scripts/basin_z_values/gen_points_in_basin.py#L85
2. https://github.com/ucgmsim/mapping/blob/deda5ebb649316ef91cca53b83a5909914017953/mapbox/vs30/scripts/basin_z_values/gen_basin_stats.py#L121

I think this makes sense to remove `wgs_nztm2000x` once I'm certain that they produce (similar) outputs.

**Comment by lispandfound** [comment]
@claudio525 regarding your question as to whether they produce the same output. The answer is _almost_. If we convert the coordinates for christchurch, then from the qcore.coordinates module we get:

```python
>>> qcore.coordinates.wgs_depth_to_nztm(np.array([-43.5320,172.6366, 0]))
array([5180040.61473068, 1570636.6812821 ,       0.        ])
```

From the geo module we instead get:
```python
>>> qcore.geo.wgs_nztm2000x(np.array([172.6366, -43.5320]))
array([[1570636.6812821 , 5180040.61451427]])
```

Aside from the obvious transposition in coordinate positions, the pyproj code and the old code agree to within $10^{-11}$ in the easting direction and completely agree in the northing direction. I think this is close enough to safely remove this old code and add a PR to the mapping repo to replace this code.

**Comment by lispandfound** [comment]
The old function is 195 lines of code in the geo module.

**Comment by sungeunbae** [comment]
I'm happy to switch to `qcore.coordinates.wgs_depth_to_nztm`

**Comment by claudio525** [comment]
Agreed, differences are tiny, should be no problem removing the old code. Nice work, great to see a reduction in code haha!

**Comment by sungeunbae** [comment]
As I'm working on a PR around the code that uses `wgs_nztm2000x`, I will convert this to the new function assuming https://github.com/ucgmsim/qcore/pull/303 is merged into master.

**Comment by lispandfound** [comment]
@sungeunbae one thing to note about the change will be that `wgs_nztm2000x` "automatically" detects if the input is in lat/lon format vs nztm format; calling `wgs_nztm2000x(wgscoords)` gives nztm coords and vice-versa. If you're relying on that behaviour be aware that the new coordinates module does not do this. Instead, you have one function `wgs_depth_to_nztm` and another `nztm_to_wgs_depth` to convert back and forth.

**Comment by sungeunbae** [comment]
> @sungeunbae one thing to note about the change will be that `wgs_nztm2000x` "automatically" detects if the input is in lat/lon format vs nztm format; calling `wgs_nztm2000x(wgscoords)` gives nztm coords and vice-versa. If you're relying on that behaviour be aware that the new coordinates module does not do this. Instead, you have one function `wgs_depth_to_nztm` and another `nztm_to_wgs_depth` to convert back and forth.

Thanks for the info :)
